### ISSUE TITLE
Issues with Lemken drill, and Agriweld muck spreader

### ISSUE THREAD

**Issue by lansalot** [Open-Agriculture/AgIsoVirtualTerminal#33]
Managed to get a fully working terminal up for a Lemken 4m drill today, was very impressed. Went to record it with OBS and it disappeared. I couldn't get it back despite clearing all program settings, removing the %APPDATA% stuff, you name it.

Some things I noticed:
- it's pretty easy to hang AVT and make it unresponsive just even trying to close the app
- more feedback during the initial terminal transfer setup would be good, even some sort of progress indicator
- there's a message that it's trying to start the CAN interface, but nothing to say it has successfully done so which gives the impression it's still trying to start it somehow

Log files attached for both devices, I caught a few, to see if anything would help in particular.

[AgISOVirtualTerminalLogs_1_Nov_2024_2_28_24pm.zip](https://github.com/user-attachments/files/17602267/AgISOVirtualTerminalLogs_1_Nov_2024_2_28_24pm.zip)
[AgISOVirtualTerminalLogs_1_Nov_2024_2_33_15pm.zip](https://github.com/user-attachments/files/17602269/AgISOVirtualTerminalLogs_1_Nov_2024_2_33_15pm.zip)
[AgISOVirtualTerminalLogs_1_Nov_2024_1_45_00pm.zip](https://github.com/user-attachments/files/17602270/AgISOVirtualTerminalLogs_1_Nov_2024_1_45_00pm.zip)
[AgISOVirtualTerminalLogs_1_Nov_2024_1_59_25pm.zip](https://github.com/user-attachments/files/17602272/AgISOVirtualTerminalLogs_1_Nov_2024_1_59_25pm.zip)
[AgISOVirtualTerminalLogs_1_Nov_2024_1_13_13pm.zip](https://github.com/user-attachments/files/17602274/AgISOVirtualTerminalLogs_1_Nov_2024_1_13_13pm.zip)

Am seriously impressed tho, this is great work! My friend checked through the Lemken screens and it appeared most of the data from the seeder was there as expected. Just needs some stability in terms of pulling the VT through in the first place,  tho this might well be something to do with the seeder itself of course. We tried reconnecting the ISOBUS entirely from the implement, but no joy.

**Comment by ad3154** [comment]
Yeah, I think the hanging issue is some kind of deadlock with the render/message thread and the CAN stack... I have been meaning to fix that. I think the updates for the stack could be run on the message thread to eliminate that issue completely.

Initial transfer - totally agree, that one is this ticket #4 

> there's a message that it's trying to start the CAN interface, but nothing to say it has successfully done so which gives the impression it's still trying to start it somehow

Good feedback, would be easy to add something there saying it worked.

I'll take a look through the logs and see if I can find out what's going on
